screens/MagicBall.js

```diff
@@ -0,0 +1,158 @@
+import React, { useState, useEffect } from 'react';
+import {
+  View,
+  Text,
+  TouchableOpacity,
+  TextInput,
+  StyleSheet,
+  Alert,
+  ImageBackground,
+} from 'react-native';
+
+import { answers } from './components/answers';
+import { MagicLoader } from '../components/MagicLoader';
+
+const image = {
+  url: 'https://media.istockphoto.com/illustrations/blue-glowing-energy-ball-on-black-background-illustration-id937266238?k=20&m=937266238&s=612x612&w=0&h=JIsUmWm9HzPbZ9LNLeJQpQWknqxCP5jdyalFE6ZNJJU=',
+};
+
+export const MagicBall = ({ navigation }) => {
+  const [magicAnswer, setMagicAnswer] = useState('');
+  const [question, setQuestion] = useState('');
+  const [loading, setLoading] = useState(false);
+  const [newQuestion, setNewQuestion] = useState('');
+
+  const generateMagicBall = () => {
+    if (question !== '' && question.length > 3) {
+      setLoading(true);
+      const answer = answers[Math.floor(Math.random() * answers.length)];
```

Comment:
First time I'm seeing randomizing function after GuessWho project, I am excited to see that you are using functions that we learned from many weeks ago!!